### DISCUSSION TITLE
[Feature] upload to album

### DISCUSSION THREAD

**Discussion by tobiashort** [immich-app/immich#12190] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hello

I don't like the automatic backup feature so it is disabled for my device. I rather like to backup only those photos which I personally care about. Those images I cherry-pick and I do upload and assign to an album. Now this is unfortunately not as simple as I would like it. 

Current procedure:
1. I select the photos
2. I click on upload (this removes the selection)
3. I have to scroll through the photos I just uploaded, reselect them.
4. I add the selected photos to the album.

Optimal solution:
1. I select the photos
2. I have a button "Add to album"
3. I select the album. Images are uploaded and directly assigned with the album. 

![2024-08-31_16-34](https://github.com/user-attachments/assets/63be564a-a440-4fc0-92fb-380559379066)

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by hrdl-github** [comment]
As pointed out in #8595 , this is also relevant for users who do synchronise their assets automatically when automatic foreground backup is running but has not yet synchronised the assets in question, e.g. because they have just been added.

**Comment by timothyfrn** [comment]
This would be awsome. I am also thinking about a upload button just like in the browser version.

**Comment by josx2** [comment]
+1 on this....this is something Synology photos does that I miss seeing in immich...super handy feature for sure!

**Comment by thebashsupremacy** [comment]
This definitely needs to be up on the prio list. Thanks!

**Comment by scoryllo** [comment]
I am also eagerly waiting for this one.
It would also help a lot with shared albums where not all participants use immich as a backup for their photos and just want to upload some of them to some album.

**Comment by Xornop** [comment]


**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by ThomasLeister** [comment]
I'm also waiting for this feature. I don't synchronize _all_ my photos, but I select the ones that I care about and create an album from them (e.g. field trip, birthday party, some other event, ...). It'd be more convenient to just be able to "upload and create an album" in one step. :muscle:

**Comment by SimJoSt** [comment]
This feature is also very relevant for people uploading media from a wide range of dates, like it happens when just migrating to Immich from a previous system.
It is a big hassle to first upload the media to the general timeline and then have to go through the whole date range and select the just uploaded media from between other already uploaded media. Especially, with high volumes of media.

That's why I mostly used the desktop upload for now.

**Comment by fett327** [comment]
Just for reference, it seems this specific feature has been asked for in many previous requests ( #11101, [#23078](https://github.com/immich-app/immich/discussions/23078), [#20892](https://github.com/immich-app/immich/discussions/20892), [#10959](https://github.com/immich-app/immich/discussions/10959), #23506).

In some cases, it gets confused as a duplicate for #1678 (which it definitely isn't, to my understanding).  I think all I'm hoping for is that the feature gets requested and doesn't get immediately closed.  This specific one is still open and running so I think devs are aware of the need (crossing fingers)

**Comment by thebashsupremacy** [comment]
Isn't this feature a basic use case? I just don't get it. Are 99% of the users activating "upload everything always"?

**Comment by LaVINCENTgit** [comment]
I'm really interested in this feature as well. It would be fantastic to have the possibility to add items directly to an album. If it helps, I think this feature is so highly requested that we would also be happy if this option were available from inside the app (for example in the 'Libraries' tab), in case it can't be added to the 'Upload to Immich' function. Thanks a lot, team.

**Comment by saarsurfer0815** [comment]
Hello dear development team,

I also really miss the photo upload feature within the Android app. This feature is what truly makes the app fully usable. At the moment, I’m still using a Synology NAS with Synology Photos, but in the mid-term I plan to switch to a Ugreen NAS. Immich will then run there via Docker, which is why I’m already testing it now.

However, this feature is so essential that, unfortunately, it’s currently a dealbreaker for me when it comes to Immich. With this feature in place, Immich would definitely have what it takes to let Synology Photos retire.

Thank you very much for all the great work you’ve done so far.

**Comment by BenjaminRatajczak** [comment]
I absolutely agree with every comment here.
I want to have access to all our (mine and my wife's) photos, but want to have separate sections for each one of us, just like a private gallery as an album. 
I just set immich up and now have to "sort" about 3000 files because I can't upload from my phone to an album and from my wife's to another directly.

Even though this is an annoying issue, my first impression of this whole project is overwhelmingly positive, and I'm excited to set everything up!

Thanks to the devs for all the awesome work!

**Comment by Xornop** [comment]
I'm frustrated that this feature is being requested many times and each time the devs just close the topic instead of listening to the community. Funny, I wanted to move away from other services because they didnt listen to their community.

**Reply by bo0tzz** [reply]
You commented this on a thread that's still open??

**Comment by oodoloo** [comment]
Hi everyone,
I'd really like to be able to use this feature and would be happy if it were expanded. I otherwise think the app is really well done, but I've really missed this feature. Especially when migrating to Immich and then having to upload all the pictures first and then move them to the appropriate albums. And then you might even have to painstakingly search for them in the timeline. So, this is a very important feature that I'm eagerly awaiting.

**Comment by thisisgonnatakeawhile** [comment]
This feature is paramount for those of us who rely on Storage Templates to organize uploaded media into a specific folder structure.

As it stands, the process involves: Uploading pictures/videos > Organizing them into albums > Using the Storage Template Migration job to store the media as needed (for example with a `{{y}}/{{MM}}{{#if album}}/{{album}}{{/if}}/{{filename}}` template). This is quite inconvenient (especially for the non tech-savvy or older people).

I hope the Immich devs can make this happen someday. 
Thanks for all your effort and dedication, you guys are the best.

**Comment by Alderaaan** [comment]
I use the app since some weeks, and this one is my first stupefaction Oo.
Hope it's will be done with the next road map...

I just don't understand why it's not in place at the beginning, but the app is very nice now for all others truffs.

**Comment by Kevin-Diepenmaat** [comment]
It is indeed cumbersome to upload photo's from the app and afterwards move them to an album. This should able to be done upon upload or select a existing album. It would be great if it can be added to the roadmap.

Thanks and keep up the great work.

**Comment by bigbroo2** [comment]
PLEASE Add this feature! It will save people thousands if not millions of hours doing double job of first backing up photos and then assigning them to the albums. I'm happy to donate $100 for this

**Comment by Xornop** [comment]
Have you guys tried Ente? Can also be selfhosted and supports this feature _and_ 2FA

**Reply by LaVINCENTgit** [reply]
Ente? No, never heard. I will read about this. thanks

**Comment by souverain-ovh** [comment]
Hello, and my apologies to the administrators of this GitHub account—I hadn’t seen this issue, so I accidentally created a duplicate. I’d still like to share my perspective here, and I hope it will be taken into consideration:  

Let me explain:
A while back, I switched to PixelUnion.eu, which uses immich, so I have to use the immich iOS app. So far, that makes sense and there’s no problem.
Except that the user experience, compared to Google Photos, is—sorry to say—awful. Let me explain, and I don’t blame anyone, but I’d like to convince the people around me to switch to immich. However, I’m stuck because the current solution is heavily criticized and is seriously not practical at all.

On the Google Photos app, once I’ve granted full access to my photo library from my iPhone, adding photos couldn’t be simpler. I go to the album in question, tap the icon at the bottom to add photos, select the photo(s), and tap “Add.”

On the Immich iOS app, it’s really not practical. It took me several tries and tests to figure it out, and even after I did, I find it takes way too long and involves too many steps—especially since user experience is a major factor if we want to convince people to widely adopt an alternative solution like Immich. And I can tell you that you really have an excellent solution and it’s a great alternative, but not in this regard.

Here are the steps I need to follow in the Immich app on iOS:
Once it’s installed and I’ve granted access to my iPhone’s photo library, I need to go to “Library” and then select “On This Device,” then select my photo(s), tap “Send,” and once my photo is sent to the Immich server (or my Immich instance—it doesn’t really matter), I have another step: I have to go to the album in question, and only then can I add the photo I just uploaded to that album.

You can imagine that, sure, it’s doable, but it’s not well thought out. I’m sorry to say this, and I don’t mean to offend anyone by saying it. On the contrary, if the developers or user experience managers of the iOS Immich app are reading this, I’m just asking them to try Google Photos (if they haven’t already—and I’m not claiming to know better than anyone else, that’s not the point of this request at all), and they’ll see, objectively speaking, that the experience offered by Google Photos is still much more user-friendly for both users and newcomers.

In short, this is a heartfelt plea, because I’d really like to be able to leave Google Photos for good, convince the people around me, and—most importantly—get my whole family to switch to this alternative. But for now, this is too frustrating, because it’s not just a matter of changing a habit—which, at the end of the day, is just part of the game and doesn’t really matter— but it’s about making sure the process isn’t more complicated—precisely to make adoption easier by being able to show that it doesn’t take any longer than before. And that’s really what the change is all about! I hope you’ll understand my heartfelt plea—it’s sincere and without reproach!

**Thank you in advance for considering my request, and I hope it will be successful!

And thank you to all the contributors to this wonderful project!**

**Reply by fett327** [reply]
Totally understand how you feel.  If I were the only one taking photos of my kids and viewing them, I would easily switch to Immich and deal with the added usage complexity.
But since my wife also takes/uploads half of our photos, it would be a strain to have her adapt to something more complex for our Album uploads.  I unfortunately have to keep throwing money at Google for extra storage until this feature gets included.

**Reply by souverain-ovh** [reply]
I hope you can cancel your Google subscription soon, like I'm trying to do; we're really very close to our goal. Hopefully, the developers will read this and understand our frustration! I believe it will happen and I'm optimistic!

**Reply by fett327** [reply]
I gotta be fair tho.  If you have ever looked at the feature request list and sorted it by "up votes" you'll see that there are quite a few features still in the queue.
Some of them deserve to be higher on the list, some of them dont (IMHO).   It would be unfair for me to expect my issues be addressed above all the others.  Especially true given the amount of my donations thus far ($0).  Just gotta be patient.

**Reply by souverain-ovh** [reply]
Patient I understand, but the request dates from August 2024 anyway!

**Comment by Xornop** [comment]
It tells us a lot about the developers that they've marked the relevant comments above as "off topic". They may be repetitive or complaining, but they _are_ on topic. Behaviour like this has pushed me away from Immich. They don't want to accept real feedback or feature request and rather just say "no we are not going to add 2FA/upload to album/etc because we dont like it" instead of doing what the community really wants.

**Reply by bo0tzz** [reply]
The only comments marked off-topic are ones talking about a different project entirely. There's a few marked duplicate because that's exactly what they are, they don't add anything to the discussion and should just be upvotes on the main post.

**Reply by souverain-ovh** [reply]
Okay, I hear and understand. However, I'm not looking for conflict; I'm just trying to help by providing feedback and trying to make it clear that this request is important for mass adoption and that we're truly at a turning point in achieving it, and I deeply believe that! Thanks again to all the contributors for their attention, and I hope we'll succeed.

**Comment by thebashsupremacy** [comment]
The thing is, for people like us commenting here, this is an absolute no brainer, it's just uncomprehensible how such a feature is not standard. I can relate to everyone complaining about it missing. But apparently there are (many) other people and perspectives also, so it seems that this feature is not as desired as others are. My guess (based on what I see in the Immich Github) is that most users just activate the "upload everything alwyas" feature and don't have to deal with this problem at all. 
But, coming from "the real world", I can assure you guys, that this feature is a BLOCKER for many non technical users to accept this software (that I run on my own private server) as a better alternative to Google Photos. 

Thanks to everybody who is involved in this project!!

**Reply by souverain-ovh** [reply]
Yes, I completely agree with you that this is a real obstacle to leaving Google Photos; it's truly the last feature that allows everyone to permanently leave Google and increase the use of immich. I also don't understand why this problem hasn't been fixed already. It's frankly a shame.

**Reply by alextran1502** [reply]
>  it's truly the last feature that allows everyone to permanently leave Google

@lelabtv You will be amazed that we see this comment for most feature requests out there 😅

**Reply by souverain-ovh** [reply]
But then why don't they do it? I don't get it!

**Reply by alextran1502** [reply]
Usually, it comes down to the balancing act of

- Manpower
- Technicality
- Maintainability
- Niche-ness
- Priority

**Reply by sudosf** [reply]
Will this feature be at least considered in the near future? i am happy to contribute to its addition as a dev, if the issue is dev capacity, i would really love to have it on the mobile app

**Reply by souverain-ovh** [reply]
I have no idea, but if you have the chance to volunteer for the team, as a non-developer user, I’d definitely say YES, and you’d make a lot of people happy!  

Thank you in advance for your time and your generous offer!  🙏

**Comment by banto6** [comment]
I noticed a relevant PR; perhaps we should await good news. https://github.com/immich-app/immich/pull/28123

**Reply by Xornop** [reply]
It has been merged!

**Reply by souverain-ovh** [reply]
Hello, sorry, I see this has been merged, but how exactly will this affect the end user? Will there be a new iOS version to benefit from this fix? I'm having trouble understanding how updates work. Thank you in advance for your reply.

**Reply by fett327** [reply]
It has been merged into the codebase.  The next time a release is built the feature is likely to be included.  This usually happens within a few days to weeks at most.
You will then have to upgrade your Immich server and app to try it out!

**Reply by souverain-ovh** [reply]
Thank you for that clarification.

**Reply by Merlijnv** [reply]
When will this be released? It has been 3+ weeks

**Reply by souverain-ovh** [reply]
now !

**Comment by Finke3** [comment]
Is this direct upload into an album also available as API?

**Comment by Xornop** [comment]
How do you even upload straight to an album? i open my gallery, open the desired picture, tap share, tap immich, and there is no option to upload to album. i thought this would be added now with 3.0.0? my app and server are updated of course.

**Reply by oodoloo** [reply]
> How do you even upload straight to an album? i open my gallery, open the desired picture, tap share, tap immich, and there is no option to upload to album. i thought this would be added now with 3.0.0? my app and server are updated of course.

I have the same problem. I tried several methods too, but couldn't manage to upload an image directly into an album. I'm a bit puzzled as well.

**Reply by souverain-ovh** [reply]
I had to wait for version 3.0.1 for this to be fixed and for it to work. It's mentioned here: https://github.com/immich-app/immich/releases/tag/v3.0.1

With version 3.0.1, it works for me (iOS)

**Reply by Xornop** [reply]
@souverain-ovh 
Im on android. Im not seeing the option. Just an upload button. App and server are 3.0.1. am i missing something?

**Reply by LaVINCENTgit** [reply]
Hello, not in this way. Go to "library" in the immich app, open your Camera folder and select what u want to upload. Below there are your albums

**Reply by bbrospo** [reply]
I imagine this is confusing many people. Me too for a while. I expected a upload button from within an album where you could browse local pictures.

**Reply by Xornop** [reply]
@LaVINCENTgit nope that also doesn't allow me to upload to an album. it just has an upload button but doesn't give an album selection option.

**Reply by LaVINCENTgit** [reply]
Have you ever seen the new screen showing new features when first app run? If not, maybe u still have prev version. Im on the 3.0.1

**Reply by oodoloo** [reply]
I’ve seen that screen, and my app also shows version 3.0.1 build 3054, but—as I mentioned—I’m still having the same problems as Xornop.

**Reply by LaVINCENTgit** [reply]
Have u tried with long press on the pic?

**Reply by Xornop** [reply]
> Have u tried with long press on the pic?
That works!

**Comment by ajonesw** [comment]
Is it just me here or does this feature still not work. I get no option to add a photo to an album when uploading from the share sheet / camera roll on iOS - only the option to upload. Then when done all I have is 
<img width="1284" height="507" alt="image" src="https://github.com/user-attachments/assets/f5fbcc31-4e5a-4447-a2bd-0f996604e6f2" />

**Reply by Xornop** [reply]
That apparently is not part of the update. Check #29979

**Reply by ajonesw** [reply]
is it just me then or does the `upload to album` feature being closed not make much sense if you can't actually upload photos to an album. After much head scratching I still don't get it. `Upload` (meaning when you add photos to immich) you should get to select an album or albums for them to be added to?

**Reply by sudosf** [reply]
From my understanding, i believe that's a whole new feature entirely as @Xornop  mentioned, this one is specific to "upload to album" inside the immich app

**Reply by canton7** [reply]
1. Select an image which hasn't been uploaded yet from the "Photos" tab
2. A sheet pops up at the bottom of the page, with "Share", "Delete from device" and "Upload"
3. Drag up the sheet to expose "Add to album", a search bar, and a list of recent albums
4. Select an album
5. Your image is uploaded, and then added to the album

**Reply by bbrospo** [reply]
It really seems like the solution should have been adding a "Upload to Album" option inside the album itself. That would have been way more simple.

**Reply by canton7** [reply]
That does actually work for me as well. I hadn't spotted that!

**Reply by ajonesw** [reply]
@canton7 maybe I am missing something here here is the `Photos` tab only shows me photos that have already been uploaded - so i can't select what hasn't been uploaded into Immich to uplaod and add to album 😯

**Reply by canton7** [reply]
@ajonesw Under the Backup settings (little cloud icon in the top right next to your user avatar), I have "Enable Backup" set to "Off", and then under "Backup Albums" I have a couple of Android's albums selected, including "Camera".

This means that all of the photos on my phone appear in Immich on my phone under "Photos", but they aren't uploaded until I select them and click "Upload" (or add them to an album, now!)

**Reply by bbrospo** [reply]
@canton7  -  on Android, may work on iphone?

1. Go to library tab, select "on this device" 
2. find a photo from library, long press on it. 
3. Scroll up the box that popped up at the bottom. 
4. Select album to upload to. 


Again It really seems like the solution should have been adding a "Upload to Album" option inside the album itself. That would have been way more simple. For users that don't backup everything, and just want to add a single recently taken photo. Adding a upload to album, from within the album would have been easy.

**Reply by canton7** [reply]
@bbrospo Fair, that might be nice.

If you have things set up as I described just above, you *can* select local-only photos from the "Add to album" section.

**Reply by ajonesw** [reply]
Appreciate the input, but this is really confusing.  So you have to add the camera Roll to backup albums - even though you don’t actually want to backup that folder.  I already have one album on my phone that I automatically back up, with Auto Backup = ON.. as I manually add photos to that album to sync them.  Presumably changing to the approach above would mean I couldn’t do both? 🫠

It feels like the app should maybe have an option to show the camera roll by default, or if this was enabled as an options, let you see which photos had been imported vs not - and maybe even which were in albums vs not

Either way I too am really confused by the entire implementation of this and feel like i just went around in circles tapping and looking what I must be doing wrong 😆

**Reply by bbrospo** [reply]
@ajonesw - try my steps above, using the library tab. 

@canton7  - Understood, but many don't want to upload everything. Just pick a single or set of photos. You could do it via the library tab, it's just not intuitive to find it there. I expect it inside the actual album.

**Reply by canton7** [reply]
@bbrospo I don't have anything which is automatically uploaded. As stated, I have "Enable Backup" set to "Off", which means that nothing is automatically uploaded, and I manually upload all photos that I care about (these days, by adding them straight to an immich album).

**Reply by bbrospo** [reply]
Ah! I see. Sorry, I misunderstood.

Try the Library tab option as well. That should work for any picture, as long as Immich has access to it.

**Reply by canton7** [reply]
@ajonesw

> It feels like the app should maybe have an option to show the camera roll by default, or if this was enabled as an options, let you see which photos had been imported vs not - and maybe even which were in albums vs not

That's how I have it set up. Nothing is automatically uploaded, and the "Photos" tab shows a mix of images which are on my phone only and those which are on immich, with a little icon showing which is which.

**Reply by canton7** [reply]
> Try the Library tab option as well. That should work for any picture, as long as Immich has access to it.

Yep, that's a different flow (which I hadn't properly appreciated).

I'm just a normal user like you, I wasn't involved in developing this, and I agree that the use-cases which have been described are probably worth addressing. I'm just explaining how using the new feature to upload an image to an album can work, currently.

**Comment by thisisgonnatakeawhile** [comment]
My current two cents:

1. UI Inconsistency in the Android app: Selecting multiple local pictures/videos (from Library > On this device) or long pressing one, reveals a sliding panel that allows the user to upload directly to an album. Conversely, selecting a single local picture/video only reveals an 'Upload Icon'. This second way of attempting an upload could also feature the 'Add to' icon that appears after selecting a photo in the photo stream, or reveal that same sliding panel that appears after selecting multiple pictures.
2. Directly uploading to an External Library is still not possible. I don't know how difficult would be to implement this feature, but in my opinion it's essential. For those interested, there's a feature request here: [5453](https://github.com/immich-app/immich/discussions/5453).
